WordPress/WordPressTest/PostListViewModelTests.swift

```diff
@@ -0,0 +1,51 @@
+import XCTest
+@testable import WordPress
+
+final class PostListViewModelTests: XCTestCase {
+    private let mockContextManager = ContextManagerMock()
+
+    func testFilterSettingsIsPostType() {
+        let sut = PostListViewModel(blog: makeBlog(), postCoordinator: PostCoordinator())
+        XCTAssertEqual(sut.filterSettings.postType, .post)
+    }
+
+    func testEditInvokesFailureWhenPostIsUploading() {
+        class MockPostCoordinator: PostCoordinator {
+            override func isUploading(post: AbstractPost) -> Bool {
+                true
+            }
+        }
+
+        let sut = PostListViewModel(blog: makeBlog(), postCoordinator: MockPostCoordinator())
+
+        var postFailedCounter = 0
+        sut.editingPostUploadFailed = {
+            postFailedCounter += 1
+        }
+        sut.edit(PostBuilder().build())
+
+        XCTAssertEqual(postFailedCounter, 1)
+    }
+
+    func testEditInvokesSuccessWhenPostIsUploading() {
+        class MockPostCoordinator: PostCoordinator {
```

Comment:
[Nice to have]: What are your thoughts about pulling this out into a private class at the bottom of the tests?

It probably isn't really necessary at the moment since it only appears twice. But if it becomes a repeated pattern, moving it to the bottom instead of declaring it in each test would make things easier to read.

Reply:
I wanted to do that, but, the issue is in every test we will need a different version of the same class. So functions will need to return different values. What we can do is to create a mock with variables that we return in those functions and modify them instead. I felt putting the class directly there makes it obvious but if there will be redundancy it would be wise to go for some other route. I will be keeping this in mind and use a common mock as soon as it makes more sense.